### ISSUE TITLE
Entries page in CMS will not fully render sometimes

### ISSUE THREAD

**Issue by ccampbell-nra** [umbraco/Umbraco.Forms.Issues#1256]
In a nutshell, in CMS v14.1.0 and UF v14.0.2 when I click the "View Entries" link on the forms start page in the CMS the entries view renders fully only some of the time. If I, instead, open the tree "Forms" folder and then click on one of my forms and then click on the "Entries" button at the top right of the screen the entries view always loads properly.

I don't know what triggers this. The site is in development and I'm running it as a debug session in VS 2022. At first the entries page will load just fine. But after a little while the page will only partially render. There are no warnings or errors listed in the browser dev console.

Reproduction
------------

If you're filing a bug, please describe how to reproduce it. Include as much
relevant information as possible, such as:

### Bug summary

There are two ways to see form entries in the CMS. One is to use the Forms tree to drill down to a form and then click the "Entries" icon. The other is shown on the default Forms page. All created forms are listed along with a green badge that shows how many entries exist. If you mouse over items in the list two links will appear: "Edit form" and "View entries".

At the start of a debug session in VS 2022 using the default page "View entries" renders the entries just fine. I am testing form submissions so I have the CMS open in one window and the front-end site open in another tab. I submit a form and then go the CMS tab and click on "Forms" to load/reload the page and then I mouse-over my form and click the "View entries" link. After a couple of form submissions this link will stop properly rendering the entries list. Instead it renders a partial header box, but no column titles, and no entries. I have attached a screen capture.

### Specifics

Umbraco CMS v14.1.0
Umbraco Forms is v14.0.2
Visual Studio 2022 Enterprise
Google Chrome v126.0.6478.127
The URL that doesn't render properly is:
https://localhost:44363/umbraco/section/forms/workspace/form/edit/f4255136-b495-4e62-a6d4-45a85763aec6/view/entries/table (contact)
or
https://localhost:44363/umbraco/section/forms/workspace/form/edit/c9d888cf-eedf-431c-b071-351992d084ec/view/entries/table (custom)
or
https://localhost:44363/umbraco/section/forms/workspace/form/edit/894a05e2-fa98-49c1-b5ce-3b08a2e086f9/view/entries/table (comment)
I have not tried other browsers.
And there are no errors displayed on the page or in the browser console.

### Steps to reproduce

Create a new VS project using the Umbraco Project template. Install Umbraco Forms via nuget packages. 

In the CMS create a basic document time to display pages; in my case it's called Web Page and it has two properties; pageTitle (textstring) and bodyContent (Richtext editor).

In the forms section create three forms. The first uses the "comment form" template. Don't make any changes, just save it. The second uses the "contact form" template. Don't make changes to this on either. The third uses the "blank" template and for mine I added one Short Answer question, three Long Answer questions, and six Single Choice questions. Most of the Single Choice questions have "Yes" and "No" prevalues.

All three Long Answer questions have conditions; the same type of conditions. A Long Answer question is displayed if a particular Single Choice question's answer is "Yes".

Create a layout template to hold any navigation menu that you want and to define the basic page HTML and call @RenderBody. Also be sure the @Html.RenderUmbracoFormDependencies(Url) call is in there.

Create templates to display content. One template to display the pageTitle and bodyContent properties. Then three similar templates but with the "await Component.InvokeAsync("Render Form")" for each form so you have one form displayed on each template. All of these templates should define the above layout template as it's Layout; ie: Layout = "layout.cshtml".

Then associate the document type with the templates so that you can then create content. Create a page for each form and set it's template to match. Publish all your pages so you can render them in a browser.

Now visit your pages and submit the forms. A couple of times for each form.

Go back to the CMS and click "Forms" in the main menu. In the start section you should see a list of forms under the headline "Your forms". If you mouse over each form in the list you'll see a "View Entries" link appear. Click this link.

Go back to the front-end site and submit a form again.

Go back to the CMS and click "Forms" in the main navigation menu. Click on the "View entries" link when it appears.

### Expected result

I expect to see the list of entries for the selected form.

### Actual result

For the first couple of times you'll get the expected list of entries. But at some point the page will stop rendering the list. Instead you'll get the header bar but no column titles. And no entries. If you refresh the page you'll get a list of entries. But if you then click "Forms" in the main navigation menu and then click a "View entries" link for a form you'll likely again get the partially rendered page.

Again, there are no warnings or errors in the browser's developer console.

<img width="1541" alt="u14-1-0_uf14-0-2_entries_page" src="https://github.com/user-attachments/assets/b8c96776-6270-486a-9298-bdcc57ad4ad6">

---
_This item has been added to our backlog AB#42652_

**Comment by AndyButland** [comment]
Thanks for the detailed report on this @ccampbell-nra.  Hopefully we've found and fixed the issue with the upcoming 14.1 release, due out next week and later today in an updated release candidate.  There wasn't an obvious reason for the error nor could we reliably replicate, but having made some clean-up and specifically in removing a duplicated data request, could no longer seem to trigger the problem.

This isn't in the RC, but for the 14.1 release we are also going to add a loading indicator here, so it's a bit clearer something is still happening if the data doesn't immediately appear.

**Comment by AndyButland** [comment]
Going to re-open this as I can see it's still happening sometimes - seemingly just on the first load after an application restart.

To replicate:

- Have a form with some entries.
- Load up the back-office, click to Forms and then "View Entries" on a the form that has some entries.
- You'll get this (no errors in the console):

![image](https://github.com/user-attachments/assets/53d8dfc8-876f-4127-8a22-6f8124c55746)

- Click "Design", then "Entries" - the entries load.
- Click back to the Forms dashboard and then "View Entries" again - the entries load.
- Click back to the Forms dashboard, do a fill refresh and then "View Entries" again - still the entries load.
- Restart the web application, do a fill refresh and then "View Entries" again - still the entries load.
- Log out and in again, go back to the Forms dashboard and then "View Entries" again - still the entries load.
- Close browser, restart web application click to Forms and then "View Entries" - get the empty table again.

**Comment by nathanwoulfe** [comment]
I'm fairly sure I've tracked down the cause, and found a solution - I was able to reproduce using Andy's steps, and with my changes can no longer reproduce.

Issue was a race condition where the CMS was requesting the collection items before Forms was able to resolve the form ID, resulting in an empty response. Juggled some pieces to await a non-empty result from the form ID observable, and all appears to be working correctly.
